docs/howto_kubeconfig.md

```diff
@@ -0,0 +1,104 @@
+# Kubeconfig management
+Every time MicroShift starts it generates a set of kubeconfig files for local and remote management of the apiserver. These are tied to the certificates for the apiserver and are also impacted by MicroShift's configuration.
+
+## Configuration
+MicroShift allows configuring additional names and IP addresses on top of the defaults, which are `<node.hostnameOverride>`, `<node.nodeIP>` and `api.<dns.baseDomain>`. A configuration example:
+```yaml
+dns:
+  baseDomain: example.com
+node:
+  hostnameOverride: "microshift-rhel9"
+  nodeIP: 10.0.0.1
+apiServer:
+  subjectAltNames:
+  - alt-name-1
+  - 1.2.3.4
+```
+All these parameters are included as common names (CN) and subject alternative names (SAN) in the external serving certificates for apiserver.
+
+## Kubeconfig files
+Upon starting, MicroShift generates a set of kubeconfig files under `/var/lib/microshift/resources/kubeadmin` for different network access. From the example above we would have:
+```bash
+/var/lib/microshift/resources/kubeadmin/
+├── kubeconfig
+├── alt-name-1
+│   └── kubeconfig
+├── 1.2.3.4
+│   └── kubeconfig
+└── microshift-rhel9
+    └── kubeconfig
+```
+Sections below will assume this configuration is loaded into the system.
+
+### Local access
+When connecting to the apiserver locally, a certificate like this one, with `localhost` as the only SAN, is served for validation:
+```bash
+Certificate:
+    Data:
+        Version: 3 (0x2)
+        Serial Number: 2 (0x2)
+        Signature Algorithm: sha256WithRSAEncryption
+        Issuer: CN = kube-apiserver-localhost-signer
+        Validity
+            Not Before: Mar 17 08:44:07 2023 GMT
+            Not After : Mar 16 08:44:08 2024 GMT
+        Subject: CN = localhost
+...
+            X509v3 Subject Alternative Name: 
+                DNS:localhost
+```
+This means we need to use `localhost` to connect to it, and have our kubeconfig use the same CA that signed the certificate above.
+
+The local access kubeconfig is always generated at `/var/lib/microshift/resources/kubeadmin/kubeconfig`. This one is not impacted/driven by configuration parameters. It allows access to apiserver using `localhost` and uses the `localhost` internal CA:
+```yaml
+clusters:
+- cluster:
+    certificate-authority-data: <base64 CA>
+    server: https://localhost:6443
+```
+This file can not be used outside of the MicroShift's host, as the CA it uses validates `localhost` network access served certificates only.
+
+### External access
+When connecting to the apiserver from an external source a certificate like this one, with all of the other names in the SAN field, is served for validation:
```

Comment:
```suggestion
When connecting to the API server from an external source, a certificate with all of the alternative names in the SAN field is served for validation, as in the following example.
```

Reply:
Done. Changed all references to apiserver in the doc too.